src/plugins/syscalls/linux.cpp

```diff
@@ -171,7 +163,7 @@ static status_t linux_build_argbuf(void* buf, vmi_instance_t vmi,
                 if ( VMI_FAILURE == vmi_read_64(vmi, &ctx, &pt_regs[i]) )
                 {
                     fprintf(stderr, "vmi_read_va(%p) failed\n", (void*)ctx.addr);
-                    goto exit;
+                    return VMI_SUCCESS;
```

Comment:
This doesn't look right, shouldn't this be a VMI_FAILURE?

Reply:
Yes. This looks as a possible bug. Refactoring made this point obvious. But I could not understand exactly what the author had in mind when he wrote this code.

I can return a VMI_FAILURE value if you also think that there should be an error response.